lib/pbbuilder.rb

```diff
@@ -58,7 +58,12 @@ def set!(field, *args, &block)
     elsif args.length == 1
       arg = args.first
       if descriptor.label == :repeated
-        if arg.respond_to?(:to_ary)
+        if arg.kind_of?(::Hash)
+          # pb.fields {"one" => "two"}
+          arg.to_h.each do |k, v|
```

Comment:
Why are you doing `kind_of?` first, and then `to_h`? Either check for `respond_to?(:to_h)` or don't do `to_h`